src/base/iterators.jl

```diff
@@ -47,43 +60,78 @@ function Base.iterate(obj::BlockIterator, iter::Int=0)
     else
         obj.cols - j * obj.xbsize
     end
-    (((i, j), (nrows, ncols)), iter+1)
+    return (((i, j), (nrows, ncols)), iter + 1)
 end
 
-struct WindowIterator
-    blockiter::BlockIterator
+struct WindowIterator{T<:Integer}
+    blockiter::BlockIterator{T}
 end
 Base.size(i::WindowIterator) = (i.blockiter.ni, i.blockiter.nj)
 Base.length(i::WindowIterator) = i.blockiter.n
-Base.IteratorSize(::Type{WindowIterator}) = Base.HasShape{2}()
-Base.IteratorEltype(::Type{WindowIterator}) = Base.HasEltype()
-Base.eltype(::WindowIterator) = Tuple{UnitRange{Int}, UnitRange{Int}}
 
-windows(raster::AbstractRasterBand) = WindowIterator(blocks(raster))
+function Base.IteratorSize(::Type{WindowIterator{T}}) where {T<:Integer}
+    return Base.HasShape{2}()
+end
+
+function Base.IteratorEltype(::Type{WindowIterator{T}}) where {T<:Integer}
+    return Base.HasEltype()
+end
+
+function Base.eltype(::WindowIterator{T})::DataType where {T<:Integer}
+    return Tuple{UnitRange{T},UnitRange{T}}
+end
 
-function Base.iterate(obj::WindowIterator, iter::Int=0)
+function windows(
+    ::Type{T},
+    raster::AbstractRasterBand,
+)::WindowIterator{T} where {T<:Integer}
+    return WindowIterator{T}(blocks(T, raster))
+end
+
+windows(raster::AbstractRasterBand)::WindowIterator{Int64} =
+    windows(Int64, raster)
+
+function Base.iterate(
+    obj::WindowIterator{T},
+    iter::T = 0,
+)::Union{Nothing,Tuple{NTuple{2,UnitRange{T}},T}} where {T<:Integer}
     handle = obj.blockiter
     next = Base.iterate(handle, iter)
     next == nothing && return nothing
```

Comment:
This could be an `===`, same check occurs more often in the codebase.

Reply:
Opened #202 for it